src/services/apiEndpoint.ts

```diff
@@ -122,25 +128,29 @@ export async function getApiDetails(apiEndpoint: string): Promise<ApiDetails> {
     throw new Error('Received invalid data from the provided API endpoint');
   }
 
-  return {url, namespace: data.namespace || 'testkube'};
+  return {url, ...data, namespace: data.namespace || 'testkube'};
 }
 
 export function useUpdateApiEndpoint(): (apiEndpoint: string) => Promise<boolean> {
   const {dispatch} = useContext(MainContext);
+  const {reset: resetClusterDetailsState, setClusterDetails} = useClusterDetailsPick('reset', 'setClusterDetails');
 
   return useMemo(
     () => async (apiEndpoint: string) => {
       const prevApiEndpoint = getApiEndpoint();
+      resetClusterDetailsState();
```

Comment:
Instead of resetting it this way, you may pass `deps` to `initializeClusterDetailsStore`:
```ts
  const [ClusterDetailsProvider] = initializeClusterDetailsStore(undefined, [apiEndpoint]);
```

Reply:
This is not working as expected

Reply:
What do you mean? What is the unexpected behavior? It should work out of the box.

Reply:
I assume that state is reseting after I apply new values. So I get empty fields in a result

Reply:
Isn't it working the same way now?

Reply:
No, I firstly reset the state, and then add set values from new endpoint

Reply:
I think that I understood what was happening. I'll take a look at it and adjust the logic.

Reply:
@rangoo94  So, can we merge this PR. Seems like a problem not really related to the feature

Reply:
The problem is in this PR, not in the mechanism of initializing the store 🙂 The initialization works fine, we have incorrectly defined it there though. I will fix it today/tomorrow 👍

Reply:
@pavloburchak, I pushed the changes:

* moved the store initialization to `App.tsx` (so it will work in Cloud easily)
* reset store with initialization dependencies

Looks like it works well for me, could you check it?

Reply:
@rangoo94 looks ok to me